paymentsheet/src/main/java/com/stripe/android/paymentsheet/paymentdatacollection/ComposeFormDataCollectionFragment.kt

```diff
@@ -16,19 +16,20 @@ import com.stripe.android.paymentsheet.StripeTheme
 import com.stripe.android.paymentsheet.forms.Form
 import com.stripe.android.paymentsheet.forms.FormViewModel
 import com.stripe.android.paymentsheet.forms.TransformFormToPaymentMethod
-import com.stripe.android.paymentsheet.specifications.FormType
+import com.stripe.android.paymentsheet.model.SupportedPaymentMethod
 import kotlinx.coroutines.flow.distinctUntilChanged
 import kotlinx.coroutines.flow.map
 
 /**
- * Fragment that displays a form for payment data collection based on the [FormType] received in the
- * arguments bundle.
+ * Fragment that displays a form for payment data collection based on the [SupportedPaymentMethod]
+ * received in the arguments bundle.
  */
 class ComposeFormDataCollectionFragment : Fragment() {
```

Comment:
Can we use a `FragmentFactory` to inject the dependencies?

Reply:
No because the `FragmentFactory` needs to know the parameters when it's created in `onCreate`, and we only know which `SupportedPaymentMethod` to pass when the payment method is selected.